mlrun/runtimes/function.py

```diff
@@ -167,6 +168,114 @@ def add_trigger(self, name, spec):
         self.spec.config[f"spec.triggers.{name}"] = spec
         return self
 
+    def from_remote_source(
+        self, source, handler="", runtime="", secrets=None,
+    ):
+        """Load nuclio function from remote source
+        :param source: a full path to the nuclio function source (code entry) to load the function from
+        :param handler: a path to the function's handler, including path inside archive/git repo
+        :param runtime: (optional) the runtime of the function (defaults to python:3.7)
+        :param secrets: a dictionary of secrets to be used to fetch the function from the source.
+               (can also be passed using env vars). options:
+               ["V3IO_ACCESS_KEY",
+               "GIT_USERNAME",
+               "GIT_PASSWORD",
+               "AWS_ACCESS_KEY_ID",
+               "AWS_SECRET_ACCESS_KEY",
+               "AWS_SESSION_TOKEN"]
+
+        Examples::
+            git:
+                ("git://github.com/org/repo#my-branch",
+                 handler="path/inside/repo#main:handler",
+                 secrets={"GIT_PASSWORD": "my-access-token"})
+            s3:
+                ("s3://my-bucket/path/in/bucket/my-functions-archive",
+                 handler="path/inside/functions/archive#main:Handler",
+                 runtime="golang",
+                 secrets={"AWS_ACCESS_KEY_ID": "some-id", "AWS_SECRET_ACCESS_KEY": "some-secret"})
+        """
+        code_entry_type = self._resolve_code_entry_type(source)
+        if code_entry_type == "":
+            raise ValueError("Couldn't resolve code entry type from source")
+
+        code_entry_attributes = {}
+
+        # resolve work_dir and handler
+        work_dir, handler = self._resolve_work_dir_and_handler(handler)
+        if work_dir != "":
+            code_entry_attributes["workDir"] = work_dir
+
+        if secrets is None:
+            secrets = {}
+
+        # set default runtime if not specified otherwise
+        if runtime == "":
+            runtime = "python:3.7"
+
+        # archive
+        if code_entry_type == "archive":
+            if source.startswith("v3io"):
+                source = f"http{source[4:]}"
+
+            v3io_access_key = secrets.get(
+                "V3IO_ACCESS_KEY", getenv("V3IO_ACCESS_KEY", "")
+            )
+            if v3io_access_key:
+                code_entry_attributes["headers"] = {
+                    "headers": {"X-V3io-Session-Key": v3io_access_key}
+                }
+
+        # s3
+        if code_entry_type == "s3":
+            try:
+                bucket, item_key = parse_s3_bucket_and_key(source)
+            except Exception as exc:
+                raise RuntimeError(f"failed to parse s3 bucket and key. {exc}")
```

Comment:
`mlrun.errors.MLRunInvalidArgumentError`

Reply:
Actually, it looks like the only place you're using it, I don't see a reason not raising inside the function itself